addOns/automation/src/main/java/org/zaproxy/addon/automation/tests/UrlPresenceTest.java

```diff
@@ -0,0 +1,494 @@
+        compilePattern(this.getData().getResponseBodyRegex(), progress);
+    }
+
+    public static LinkedHashMap<?, ?> paramsToData(
+            String name,
+            String onFail,
+            String operator,
+            String url,
+            String requestHeaderRegex,
+            String requestBodyRegex,
+            String responseHeaderRegex,
+            String responseBodyRegex) {
+        LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+        map.put(PARAM_NAME, name);
+        map.put(PARAM_ON_FAIL, onFail);
+        map.put(PARAM_OPERATOR, operator);
+        map.put(PARAM_URL, url);
+        map.put(PARAM_REQUEST_HEADER_REGEX, requestHeaderRegex);
+        map.put(PARAM_REQUEST_BODY_REGEX, requestBodyRegex);
+        map.put(PARAM_RESPONSE_HEADER_REGEX, responseHeaderRegex);
+        map.put(PARAM_RESPONSE_BODY_REGEX, responseBodyRegex);
+        return map;
+    }
+
+    public UrlPresenceTest(
+            String name,
+            String onFail,
+            String operator,
+            String url,
+            String requestHeaderRegex,
+            String requestBodyRegex,
+            String responseHeaderRegex,
+            String responseBodyRegex,
+            AutomationJob job,
+            AutomationProgress progress) {
+        this(
+                paramsToData(
+                        name,
+                        onFail,
+                        operator,
+                        url,
+                        requestHeaderRegex,
+                        requestBodyRegex,
+                        responseHeaderRegex,
+                        responseBodyRegex),
+                job,
+                progress);
+    }
+
+    public UrlPresenceTest(AutomationJob job, AutomationProgress progress)
+            throws IllegalArgumentException {
+        super("", AbstractAutomationTest.OnFail.INFO.name(), job);
+        data = new Data(this);
+        data.setOnFail(AbstractAutomationTest.OnFail.INFO);
+    }
+
+    @Override
+    public String getTestType() {
+        return TEST_TYPE;
+    }
+
+    public static boolean isNullOrEmpty(String str) {
+        return str == null || str.isEmpty();
+    }
+
+    private Pattern compilePattern(String regex, AutomationProgress progress) {
+        if (!isNullOrEmpty(regex)) {
+            try {
+                return Pattern.compile(regex);
+            } catch (PatternSyntaxException e) {
+                progress.warn(
+                        Constant.messages.getString(
+                                "automation.tests.url.badRegex",
+                                getJobType(),
+                                regex,
+                                getName(),
+                                e.getMessage()));
+            }
+        }
+        return null;
+    }
+
+    public boolean matchPattern(
+            String regex, HttpMessage msg, String param, AutomationProgress progress) {
```

Comment:
Theres no point in passing both the msg and the param in here - just pass in the relevant string from the msg then you dont need the switch statement